src/quo/components/wallet/network_link/view.cljs

```diff
@@ -1,82 +1,151 @@
+             :x1             "72.271"
+             :x2             "82.385"
+             :y1             "5"
+             :y2             "34.155"
+             :gradient-units "userSpaceOnUse"}
+            [svg/stop {:stop-color (colors/resolve-color destination theme)}]
+            [svg/stop {:offset "1" :stop-color (colors/resolve-color source theme)}]]]]
+         [rn/view {:style style/bottom-left-circle-container}
+          [circle fill-color source-color]]
+         [rn/view {:style style/top-right-circle-container}
+          [circle fill-color destination-color]]]))))
 
 (defn link-2x
-  [{:keys [source destination theme]}]
-  [svg/svg
-   {:width "73" :height "122" :viewBox "0 0 73 122" :fill "none" :xmlns "http://www.w3.org/2000/svg"}
-   [svg/path
-    {:d
-     "M67.9999 5L58.6356 5C47.5899 5 38.6356 13.9543 38.6356 25L38.6356 97C38.6356 108.046 29.6813 117 18.6356 117L5.00006 117"
-     :stroke "url(#gradient)"}]
-   [svg/circle
-    {:cx     "68"
-     :cy     "5"
-     :r      "4"
-     :fill   (colors/theme-colors colors/white colors/neutral-90 theme)
-     :stroke (colors/resolve-color destination theme)}]
-   [svg/circle
-    {:cx     "5"
-     :cy     "117"
-     :r      "4"
-     :fill   (colors/theme-colors colors/white colors/neutral-90 theme)
-     :stroke (colors/resolve-color source theme)}]
-   [svg/defs
-    [svg/linear-gradient
-     {:id            "gradient"
-      :x1            "72.2711"
-      :y1            "5.00001"
-      :x2            "102.867"
-      :y2            "49.0993"
-      :gradientUnits "userSpaceOnUse"}
-     [svg/stop {:stop-color (colors/resolve-color destination theme)}]
-     [svg/stop {:offset "1" :stop-color (colors/resolve-color source theme)}]]]])
+  []
+  (let [container-width (reagent/atom 100)
+        stroke-color    "url(#gradient)"]
+    (fn [{:keys [source destination theme]}]
+      (let [source-color      (colors/resolve-color source theme)
+            destination-color (colors/resolve-color destination theme)
+            fill-color        (colors/theme-colors colors/white colors/neutral-90 theme)
+            view-box          (str "0 0 " @container-width " 114")
+            side-lines-path   (helpers/calculate-side-lines-path-2x @container-width)
+            central-transform (helpers/calculate-transform @container-width)]
+        [rn/view
+         {:style     style/link-2x-container
+          :on-layout #(reset! container-width
+                        (oget % :nativeEvent :layout :width))}
+         [svg/svg
+          {:xmlns    "http://www.w3.org/2000/svg"
+           :height   "100%"
+           :width    "100%"
+           :view-box view-box
+           :fill     :none}
+          [svg/path
+           {:d      (:left side-lines-path)
+            :stroke source-color}]
+          [svg/path
+           {:d
+            "M62.9999 1L53.6356 1C42.5899 1 33.6356 9.9543 33.6356 21L33.6356 93C33.6356 104.046 24.6813 113 13.6356 113L5.71778e-05 113"
```

Comment:
SVG's confuse the hell out of me 😅 . Is there any way to comment the general idea behind these line-1x and line2x views to give some understanding of how this should work, or how you came up with this? 🙏

Reply:
It's basically the `d` value from the exported SVG from Figma. We need to use it as a base and extend it by code to make it stretchable.